# MOMS-PI_WMTS_2025
Code used to process and analyze all vaginal metatranscriptomic, 16s rRNA vaginal microbiome profile, and vaginal cytokine data from the MOMS-PI project.

*All data in this analysis was generated by the authors of the following publication*  
Fettweis JM, Serrano MG, Brooks JP, et al. The vaginal microbiome and preterm birth. Nat Med. 2019;25(6):1012-1021. doi:10.1038/s41591-019-0450-2

*Associated Publication*  
Cunningham, M et al 2026

**Acknowledgements**
+ High Performance Computing resources provided by the High Performance Research Computing (HPRC) core facility at Virginia Commonwealth University (https://hprc.vcu.edu) were used for conducting this research.
+ This research was [in part] supported by CTSA award No. UM1TR004360 from the National Center for Advancing Translational Sciences. Its contents are solely the responsibility of the authors and do not necessarily represent official views of the National Center for Advancing Translational Sciences or the National Institutes of Health.
+ Generative AI Disclosure: GitHub CoPilot 2025 release was used for code development including data visualization. CoPilot did not influence the studyâ€™s scientific methodology, data analysis, or conclusions but were employed to improve clarity and streamline computational processes.



**Metatranscriptomic Data Processing and Analysis Requirements**

*Processing*
1. Fastp(v0.23.4)
2. Kraken2(v2.1.3 with k2_pluspf (2023-10-09) and additional in-house reference vaginal microbe genomes)
3. MultiQC(v1.17)
4. Salmon(v1.10.1)
5. R Studio(v4.4.0 or higher)
6. R packages
   + attached base packages: stats4, stats, ,graphics, grDevices, utils, datasets, methods
7. other attached R packages and versions:
   + readr_2.1.5
   + rtracklayer_1.66.0
   + tibble_3.2.1
   + dplyr_1.1.4
   + org.Hs.eg.db_3.20.0
   + AnnotationDbi_1.68.0
   + DESeq2_1.46.0
   + SummarizedExperiment_1.36.0
   + Biobase_2.66.0
   + MatrixGenerics_1.18.1
   + matrixStats_1.5.0
   + GenomicRanges_1.58.0
   + GenomeInfoDb_1.42.1
   + IRanges_2.40.1
   + S4Vectors_0.44.0
   + BiocGenerics_0.52.0
   + tximport_1.34.0

*DESeq2 analysis*

attached base packages:
 + grid
 + stats4
 + stats
 + graphics
 + grDevices
 + utils
 + datasets
 + methods
 + base     

other attached packages:
 [1] glmnet_4.1-10               Matrix_1.7-3                caret_7.0-1                 lattice_0.22-6             
 [5] pROC_1.18.5                 ROCR_1.0-11                 ggVennDiagram_1.5.4         FactoMineR_2.11            
 [9] car_3.1-3                   carData_3.0-5               variancePartition_1.38.0    DT_0.33                    
[13] fgsea_1.34.0                clusterProfiler_4.16.0      Maaslin2_1.22.0             egg_0.4.5                  
[17] enrichplot_1.28.2           magick_2.8.7                ggcorrplot_0.1.4.1          rcompanion_2.5.0           
[21] DescTools_0.99.60           plotly_4.11.0               MASS_7.3-65                 sparseLDA_0.1-9            
[25] lubridate_1.9.4             forcats_1.0.0               stringr_1.5.1               purrr_1.0.4                
[29] readr_2.1.5                 tidyverse_2.0.0             ggrepel_0.9.6               gridExtra_2.3              
[33] reshape_0.8.10              rnaseqGene_1.32.0           fission_1.28.0              RUVSeq_1.42.0              
[37] edgeR_4.6.2                 limma_3.64.1                EDASeq_2.42.0               ShortRead_1.66.0           
[41] GenomicAlignments_1.44.0    Rsamtools_2.24.0            Biostrings_2.76.0           XVector_0.48.0             
[45] sva_3.56.0                  BiocParallel_1.42.1         mgcv_1.9-3                  nlme_3.1-168               
[49] Gviz_1.52.0                 org.Hs.eg.db_3.21.0         AnnotationDbi_1.70.0        genefilter_1.90.0          
[53] ggbeeswarm_0.7.2            glmpca_0.2.0                PoiClaClu_1.0.2.1           RColorBrewer_1.1-3         
[57] pheatmap_1.0.13             hexbin_1.28.5               vsn_3.76.0                  apeglm_1.30.0              
[61] DESeq2_1.48.1               magrittr_2.0.3              tximeta_1.26.1              airway_1.28.0              
[65] SummarizedExperiment_1.38.1 Biobase_2.68.0              GenomicRanges_1.60.0        GenomeInfoDb_1.44.0        
[69] IRanges_2.42.0              S4Vectors_0.46.0            BiocGenerics_0.54.0         generics_0.1.4             
[73] MatrixGenerics_1.20.0       matrixStats_1.5.0           BiocStyle_2.36.0            devEMF_4.5-1               
[77] Rtsne_0.17                  GUniFrac_1.8                patchwork_1.3.1             tidyr_1.3.1                
[81] tibble_3.3.0                dplyr_1.1.4                 reshape2_1.4.4              ggplot2_3.5.2              
[85] vegan_2.7-1                 permute_0.9-8               phyloseq_1.52.0            




**Vaginal V1-V3 16s rRNA Microbiome Profiling Requirements** 
1. R Studio(v4.4.0 or higher)
2. R packages
   + attached base packages:
[1] grid      stats4    stats     graphics  grDevices
[6] utils     datasets  methods   base     

other attached packages:

 [1] glmnet_4.1-10               Matrix_1.7-3               
 [3] caret_7.0-1                 lattice_0.22-6             
 [5] pROC_1.18.5                 ROCR_1.0-11                
 [7] ggVennDiagram_1.5.4         FactoMineR_2.11            
 [9] car_3.1-3                   carData_3.0-5              
[11] variancePartition_1.38.0    DT_0.33                    
[13] fgsea_1.34.0                clusterProfiler_4.16.0     
[15] Maaslin2_1.22.0             egg_0.4.5                  
[17] enrichplot_1.28.2           magick_2.8.7               
[19] ggcorrplot_0.1.4.1          rcompanion_2.5.0           
[21] DescTools_0.99.60           plotly_4.11.0              
[23] MASS_7.3-65                 sparseLDA_0.1-9            
[25] lubridate_1.9.4             forcats_1.0.0              
[27] stringr_1.5.1               purrr_1.0.4                
[29] readr_2.1.5                 tidyverse_2.0.0            
[31] ggrepel_0.9.6               gridExtra_2.3              
[33] reshape_0.8.10              rnaseqGene_1.32.0          
[35] fission_1.28.0              RUVSeq_1.42.0              
[37] edgeR_4.6.2                 limma_3.64.1               
[39] EDASeq_2.42.0               ShortRead_1.66.0           
[41] GenomicAlignments_1.44.0    Rsamtools_2.24.0           
[43] Biostrings_2.76.0           XVector_0.48.0             
[45] sva_3.56.0                  BiocParallel_1.42.1        
[47] mgcv_1.9-3                  nlme_3.1-168               
[49] Gviz_1.52.0                 org.Hs.eg.db_3.21.0        
[51] AnnotationDbi_1.70.0        genefilter_1.90.0          
[53] ggbeeswarm_0.7.2            glmpca_0.2.0               
[55] PoiClaClu_1.0.2.1           RColorBrewer_1.1-3         
[57] pheatmap_1.0.13             hexbin_1.28.5              
[59] vsn_3.76.0                  apeglm_1.30.0              
[61] DESeq2_1.48.1               magrittr_2.0.3             
[63] tximeta_1.26.1              airway_1.28.0              
[65] SummarizedExperiment_1.38.1 Biobase_2.68.0             
[67] GenomicRanges_1.60.0        GenomeInfoDb_1.44.0        
[69] IRanges_2.42.0              S4Vectors_0.46.0           
[71] BiocGenerics_0.54.0         generics_0.1.4             
[73] MatrixGenerics_1.20.0       matrixStats_1.5.0          
[75] BiocStyle_2.36.0            devEMF_4.5-1               
[77] Rtsne_0.17                  GUniFrac_1.8               
[79] patchwork_1.3.1             tidyr_1.3.1                
[81] tibble_3.3.0                dplyr_1.1.4                
[83] reshape2_1.4.4              ggplot2_3.5.2              
[85] vegan_2.7-1                 permute_0.9-8              
[87] phyloseq_1.52.0            




**qPCR statistical tests**

**Cytokine Data Analysis Requirements**
